c-bindings-gen/src/types.rs

```diff
@@ -211,6 +234,16 @@ impl<'a> GenericTypes<'a> {
 					return Some(res);
 				}
 			}
+		} else {
+			let mut it = path.segments.iter();
+			if path.segments.len() == 2 && format!("{}", it.next().unwrap().ident) == "Self" {
```

Comment:
Perhaps a one-line comment spelling out what case this handling would be useful.